allennlp/modules/token_embedders/bidirectional_language_model_token_embedder.py

```diff
@@ -0,0 +1,170 @@
+            if dataset_reader_config.get("type") == "multiprocess":
+                dataset_reader_config = dataset_reader_config.get("base_reader")
+            token_indexer_config = dataset_reader_config.get("token_indexers").get(self._token_name)
+            token_indexer: TokenIndexer = TokenIndexer.from_params(token_indexer_config)
+            token_list = [Token(token) for token in bos_eos_tokens]
+            bos_eos_indices = token_indexer.tokens_to_indices(token_list, self._lm.vocab, "key")["key"]
+            self._bos_indices = torch.Tensor(bos_eos_indices[0])
+            self._eos_indices = torch.Tensor(bos_eos_indices[1])
+        else:
+            self._bos_indices = None
+            self._eos_indices = None
+
+        if dropout:
+            self._dropout = torch.nn.Dropout(dropout)
+        else:
+            self._dropout = lambda x: x
+
+        self._remove_bos_eos = remove_bos_eos
+        num_layers = self._lm._contextualizer.num_layers + 1
+        self._scalar_mix = ScalarMix(mixture_size=num_layers, do_layer_norm=False, trainable=True)
+
+        character_dim = self._lm._text_field_embedder.get_output_dim()
+        contextual_dim = self._lm._contextualizer.get_output_dim()
+
+        if not contextual_dim % character_dim == 0 or character_dim > contextual_dim:
+            raise RuntimeError(
+                    "The output dimensions for the text_field_embedder " +
+                    f"({character_dim}) and the contextualizer ({contextual_dim})" +
+                    f" from the language model loaded from {archive_file} are " +
+                    "not compatible. Please check the config used to train that " +
+                    "model and ensure that the output dimension of the " +
+                    "text_field_embedder divides the output dimension of the " +
+                    "contextualizer.")
+        self._character_embedding_duplication_count = contextual_dim // character_dim
+
+        for param in self._lm.parameters():
+            param.requires_grad = requires_grad
+
+    def get_output_dim(self) -> int:
+        return self._lm._contextualizer.get_output_dim() # pylint: disable=protected-access
+
+    def forward(self,  # type: ignore
+                inputs: torch.Tensor) -> Dict[str, torch.Tensor]:
+        """
+        Parameters
+        ----------
+        inputs: ``torch.Tensor``
+            Shape ``(batch_size, timesteps, ...)`` of token ids representing the current batch.
+            These must have been produced using the same indexer the LM was trained on.
+
+        Returns
+        -------
+        The bidirectional language model representations for the input sequence, shape
+        ``(batch_size, timesteps, embedding_dim)``
+        """
+        # pylint: disable=arguments-differ
+        if self._bos_indices is not None:
+            mask = get_text_field_mask({"": inputs})
+            inputs, _ = add_sentence_boundary_token_ids(
```

Comment:
The mask also needs to be replaced as the second return argument from `add_sentence_boundary_token_ids`